compose/utils.py

```diff
@@ -49,6 +49,9 @@ def split_buffer(stream, splitter=None, decoder=lambda a: a):
     splitter = splitter or line_splitter
     buffered = six.text_type('')
 
+    if type(stream) is bytes:
+        stream = stream.decode("utf-8")
```

Comment:
This should already be handled by `stream_as_text()`.

This also seems wrong because string should never be bytes, it should be an iterator.